modules/plugin/jdbc/jdbc-h2/src/main/java/org/geotools/data/h2/H2FilterToSQL.java

```diff
@@ -217,7 +218,7 @@ protected void writeLiteral(Object literal) throws IOException {
         if (literal instanceof Date) {
             out.write("PARSEDATETIME(");
             if (literal instanceof java.sql.Date) {
-                out.write("'" + DATE_FORMAT.format(literal) + "', 'yyyy-MM-dd'");
+                out.write("'" + DATE_FORMAT.format(literal) + "', 'yyyy-MM-dd', '"+Locale.getDefault()+"', 'GMT'");
```

Comment:
Reading that more carefully - I'm not sure what the locale does there, in my local it just writes the date out in GMT, because sqlDates have a timezone (sigh).